plugins/data_coverage/src/openjdk/jcov/data/Env.java

```diff
@@ -24,52 +24,90 @@
  */
 package openjdk.jcov.data;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
 import java.lang.reflect.InvocationTargetException;
 import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Map;
-import java.util.Objects;
-
-import static java.util.stream.Collectors.toMap;
+import java.util.Properties;
+import java.util.Set;
 
 /**
  * Much of the functionality in this plugin is controlled through system properties. This class defines some shortcuts
  * which makes it a bit easier.
  */
 public class Env {
 
-    private static Map<String, String> properties = System.getProperties().entrySet().stream()
-            .collect(toMap(Object::toString, Objects::toString));
+    public static final String PROP_FILE =
+            Env.class.getPackageName().replace('.', '/') +
+            "/instrumentation.properties";
+    /**
+     * Prefix for all system property names which will be passed to the VM running JCov calls.
+     */
+    public static final String JCOV_DATA_ENV_PREFIX = "jcov.data.";
+
+    public static final String JCOV_DATA_SAVER = "jcov.data.saver";
+    private static final String JCOV_OPTION_FOR_DATA_SAVER = "jcov.data-saver";
 
-    public static Map<String, String> properties() {
-        return properties;
+    private static volatile Properties SAVED;
+
+    public static void setSystemProperties(Map<String, String> properties) {
+        properties.forEach((k, v) -> System.setProperty(k, v));
     }
 
-    public static void properties(Map<String, String> properties) {
-        Env.properties = properties;
+    public static void clear(String prefix) {
+        Set<String> keys = System.getProperties().stringPropertyNames();
+        keys.stream().filter(k -> k.startsWith(prefix))
+                .forEach(k -> System.clearProperty(k));
```

Comment:
It's a minor comment. Just to say that here, I mean there are 2 kinds of properties -  Environment > export JCOV_OPTION=path && System.getenv("JCOV_OPTION") and java -DJCOV_OPTION=path && System.getProperty("JCOV_OPTION") First and second ways to get external properties don't have in mind their update/clean. If System properties are used as global mutable variables of the program then it would be much better to use a special singleton container like a hash table for this need. If these options are read-only then all updating/cleaning methods are superfluous.

Reply:
In fact, in the current state of the code, system properties are changed only from tests. Tests simply do what a user is suggested to do: set some system properties, run a JCov empty point class, either from command line or from some other java code. I think this is acceptable. There can be an additional interface which does not require setting system properties, but system properties will always work anyway.
There is one exception when a system property is set not from a test: "jcov.data-saver" is set to be the same as "jcov.data.saver. I am looking if this is justified.

Reply:
Upon further review, no, this is necessary. If I want to be able to implant a name of the saver SPI class into the instrumented code (into a property file), I will later have to retrieve it from the storage (the property file) and set system property, because this is where JCov is obtaining it from - from the system property. The only way to go around that would be to add some SPI to JCov itself, which would allow setting JCov up. This may be a valuable enhancement, but it will have to be done separately.